smdebug/tensorflow/keras.py

```diff
@@ -600,13 +600,20 @@ def _save_layer_input_and_outputs(self):
 
         for layer_name in self.saved_layers:
             # Save Input
-            tensor = self.saved_layers[layer_name].layer_input
-            export_name = get_export_name_for_keras(layer_name, tensor_type="input", tensor=tensor)
             input_collection = (
                 {self.get_collection(CollectionKeys.LAYERS)}
                 if self._is_collection_being_saved_for_step(CollectionKeys.LAYERS)
                 else set()
             )
+            if bool(input_collection):
```

Comment:
Can we make this check better? If `include_collection` is always a set then we can check if the size of the set is nonzero?